iothub/device/src/ConnectionStatus.cs

```diff
@@ -32,7 +32,7 @@ public enum ConnectionStatus
         /// The device connection was closed gracefully.
         /// </summary>
         /// <remarks>
-        /// To perform more operations on the client, call <see cref="IotHubDeviceClient.Dispose()"/> and then re-initialize the client.
+        /// To perform more operations on the client, call <see cref="InternalClient.Dispose()"/> and then re-initialize the client.
```

Comment:
Maybe InternalClient is the wrong name now. It feels weird to expose that type name to users.

Reply:
Also, even though it is inherited, shouldn't there be an override on each of these classes that calls the base?

Reply:
I feel the same way. ```IotHubBaseClient```?

Reply:
I agree that internal client here feels wrong, however, the doc comments aren't able to resolve the inherited method. We aren't overridding the implementations in the derived class; the device client and module client instance will call the implementations in the base internal client class for the "common" APIs.

Reply:
We're using the suggested pattern for dispose, so never mind it being an override. I still don't love the name InternalClient being what is shown to users, though. It feels misleading when they are calling it on their client instance.